data/db-schema/src/main/resources/net/corda/db/schema/vnode-vault/migration/ledger-utxo-creation-v5.1.xml

```diff
@@ -3,18 +3,106 @@
+    <property name="json.column.type" value="CLOB" dbms="hsqldb"/>
+
     <changeSet author="R3.Corda" id="ledger-utxo-creation-v5.1">
-       <createIndex
-           indexName="utxo_visible_transaction_state_idx_consumed"
-           tableName="utxo_visible_transaction_state">
-           <column name="consumed"/>
-       </createIndex>
 
         <createIndex
                 indexName="utxo_transaction_output_idx_type"
                 tableName="utxo_transaction_output">
             <column name="type"/>
         </createIndex>
+
+        <!-- Add new status column to utxo_transaction -->
+        <addColumn tableName="utxo_transaction">
+            <column name="status" type="VARCHAR(255)">
+                <constraints nullable="true"/>
+            </column>
+            <column name="updated" type="TIMESTAMP">
+                <constraints nullable="true"/>
+            </column>
+        </addColumn>
+
+        <!-- Migrate data from the old status table -->
+        <update tableName="utxo_transaction">
+            <column name="status"
+                    valueComputed="(select status from utxo_transaction_status uts where uts.transaction_id=id)"/>
+            <column name="updated"
+                    valueComputed="(select updated from utxo_transaction_status uts where uts.transaction_id=id)"/>
+        </update>
+
+        <!-- Drop the old status table -->
+        <dropTable tableName="utxo_transaction_status"/>
+
+        <!-- After status data migration we can make the status/updated column non-nullable -->
+        <addNotNullConstraint tableName="utxo_transaction" columnName="status" columnDataType="VARCHAR(255)"/>
+        <addNotNullConstraint tableName="utxo_transaction" columnName="updated" columnDataType="TIMESTAMP"/>
+
+        <!-- Add new custom_representation and consumed columns to utxo_transaction_output -->
+        <addColumn tableName="utxo_transaction_output">
+            <column name="custom_representation" type="${json.column.type}">
+                <constraints nullable="true"/>
+            </column>
+            <column name="consumed" type="TIMESTAMP">
+                <constraints nullable="true"/>
+            </column>
+        </addColumn>
+
+        <!--
+        Migrate data from the old visible states table
+        NOTE: We can't use <update> here because both tables have the same column naming
+        (transaction_id, group_idx etc.) and the SQL statement generated will not work in
+        Postgres, and we have no way to alias the base table.
+        -->
+        <sql>
+            UPDATE utxo_transaction_output txo
+            SET
+            custom_representation = (
+            SELECT custom_representation from utxo_visible_transaction_state vts
+            WHERE vts.transaction_id=txo.transaction_id
+            AND vts.group_idx=txo.group_idx
+            AND vts.leaf_idx=txo.leaf_idx
```

Comment:
Might be a bit more readable to indent this one level (same for `consumed`)

Reply:
Good point, thanks!